python/lsst/ts/ess/csc/data_client/young_32400_weather_station_data_client.py

```diff
@@ -629,57 +625,35 @@ def restart_rain_stopped_timer(self) -> None:
         self.rain_stopped_timer_task.cancel()
         self.rain_stopped_timer_task = asyncio.create_task(self.rain_stopped_timer())
 
-    async def run(self) -> None:
-        self.read_loop_task.cancel()
-        self.read_loop_task = asyncio.create_task(self.read_loop())
+    async def perform_pre_read_loop_tasks(self) -> None:
+        # Start the "rain stopped" timer so we can report "no rain"
+        # as soon after starting as it is safe to do so.
+        self.restart_rain_stopped_timer()
 
-    async def read_loop(self) -> None:
+    async def read_data(self) -> None:
         """Read raw data from the weather station.
 
         The format is as described by by DATA_REGEX.
         """
-        # Start the "rain stopped" timer so we can report "no rain"
-        # as soon after starting as it is safe to do so.
-        self.restart_rain_stopped_timer()
+        assert self.client is not None  # make mypy happy
```

Comment:
Could you please fix my existing mistake at line 636 in the doc string: "by by" -> "by"

Reply:
Thank you.